src/property.ts

```diff
@@ -896,6 +896,8 @@ export function Property$pendingInit(obj: Entity, prop: Property, value: boolean
 
 function Property$subArrayEvents(obj: Entity, property: Property, array: ObservableArray<any>): void {
 	array.changed.subscribe(function (args) {
+		Property$pendingInit(obj, property, false);
```

Comment:
Set pending init to false when an array is directly modified (ex: calling a method like `splice`)